x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -531,6 +533,166 @@ class AgentPolicyService {
     };
   }
 
+  public async bulkUpdate(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient,
+    agentPolicies: Array<Partial<AgentPolicy> & { id: string }>,
+    options: {
+      user?: AuthenticatedUser;
+      force?: boolean;
+      skipValidation?: boolean;
+      spaceId?: string;
+      authorizationHeader?: HTTPAuthorizationHeader | null;
+      removeProtection?: boolean;
+      bumpRevision?: boolean;
+    }
+  ): Promise<{
+    updatedPolicies: AgentPolicy[] | null;
+    failedPolicies: Array<{ agentPolicy: Partial<AgentPolicy>; error: Error | SavedObjectError }>;
+  }> {
+    const logger = appContextService.getLogger();
+    logger.info(`Starting bulk update of agent policies: ${agentPolicies.map(({ id }) => id)}`);
+
+    const existingAgentPolicies = await this.list(soClient, {
+      perPage: SO_SEARCH_LIMIT,
+      withPackagePolicies: true,
+    });
+
+    this.checkNameUniquenessBulk(agentPolicies, existingAgentPolicies.items);
+    this.checkPolicyExistsBulk(agentPolicies, existingAgentPolicies.items);
+
+    const policiesToUpdate: Array<Partial<AgentPolicy> & { id: string }> = [];
+    const failedPolicies: Array<{
+      agentPolicy: Partial<AgentPolicy>;
+      error: Error | SavedObjectError;
+    }> = [];
+
+    await pMap(agentPolicies, async (agentPolicy) => {
```

Comment:
should there be a concurrency limit in pMap?

Reply:
Hmm, that's a good question. This is the pattern we follow in the package policy bulkUpdate with no defined concurrency (i.e., all the promises are created concurrently). However, I notice that in other places in Fleet, it revolves around 20.

Reply:
Yes, we try to limit concurrency to avoid using too much memory at once if there are many policies.

Reply:
Should I follow the pattern and set it to 20 then?